public/app/plugins/datasource/loki/getDerivedFields.ts

```diff
@@ -22,12 +22,39 @@ export function getDerivedFields(dataFrame: DataFrame, derivedFieldConfigs: Deri
     throw new Error('invalid logs-dataframe, string-field missing');
   }
 
-  lineField.values.forEach((line) => {
+  const labelFields = dataFrame.fields.find((f) => f.type === FieldType.other && f.name === 'labels');
+
+  for (let i = 0; i < lineField.values.length; i++) {
     for (const field of newFields) {
+      if (
+        config.featureToggles.lokiDerivedFieldsFromLabels &&
+        derivedFieldsGrouped[field.name][0].labelMatcher &&
+        labelFields
+      ) {
+        const label = labelFields.values[i];
+        if (label) {
+          const intersectingKey = Object.keys(label).find(
```

Comment:
Could you please add comment here, I was getting lost. It took mw a while to understand that `matcherRegex` is label name. So maybe a comment of what are we doing here would be helpful in the future.

Reply:
Good point, will ping you once that's done!

Reply:
Done in 24062237a952e301636f37b45a8c4e021bdf8905